src/libsyntax/ast.rs

```diff
@@ -1860,7 +1860,10 @@ pub type Variant = Spanned<Variant_>;
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub enum UseTreeKind {
     /// `use prefix` or `use prefix as rename`
-    Simple(Option<Ident>),
+    ///
+    /// The extra `NodeId`s are for HIR lowering, when additional statements are created for each
+    /// namespace.
+    Simple(Option<Ident>, NodeId, NodeId),
```

Comment:
Is this really necessary?
Can't we create HIR items with fresh `NodeId`s "out of thin air"?

Reply:
While possible, we'd need to insert a cache for the `NodeId` of the `Simple` variant to the generated `NodeId`s and carry that one around in lowering.

The reason for that is that you need the same ids in https://github.com/rust-lang/rust/pull/51425/files#diff-64b696b0ef6ad44140e973801ed82b25R2742 and https://github.com/rust-lang/rust/pull/51425/files#diff-64b696b0ef6ad44140e973801ed82b25R2409

Reply:
I tried creating the `NodeId`s during lowering, but they weren't referenced by the module after lowering, so type collection hit an error when it tried to account for these items. Unless there's an easier way to reference the new `Item`s and their IDs when the module's `ItemId`s are lowered, this is what @oli-obk came up with.